<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="A.DeMartin" />

<meta name="date" content="2025-04-16" />

<title>Fbhighres</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">HumanHeartCarTrans2</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Fbhighres</h1>
<h4 class="author">A.DeMartin</h4>
<h4 class="date">2025-04-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-07-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>HumanHeartCarTrans2/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250218code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250218)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250218code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250218)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/Human_heart_allmerged_seurat.rds
</td>
<td style="text-align:left;">
data/Human_heart_allmerged_seurat.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/Human_heart_AllFb.rds
</td>
<td style="text-align:left;">
data/Human_heart_AllFb.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/markerGenesAllFb_clusterNameAllFb
</td>
<td style="text-align:left;">
data/markerGenesAllFb_clusterNameAllFb
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/Human_heart_IntFb.rds
</td>
<td style="text-align:left;">
data/Human_heart_IntFb.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinHumanHeartCarTrans2treea9a9162055dbf59e8833298a961a773e821f2ffftargetblanka9a9162a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/angeldemartin/HumanHeartCarTrans2/tree/a9a9162055dbf59e8833298a961a773e821f2fff" target="_blank">a9a9162</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomangeldemartinHumanHeartCarTrans2treea9a9162055dbf59e8833298a961a773e821f2ffftargetblanka9a9162a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/angeldemartin/HumanHeartCarTrans2/tree/a9a9162055dbf59e8833298a961a773e821f2fff" target="_blank">a9a9162</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    data/seurat files/

Untracked files:
    Untracked:  analysis/slingshotAllFb.Rmd
    Untracked:  analysis/slingshotIntFb.Rmd

Unstaged changes:
    Modified:   analysis/BMP_HGF.Rmd
    Modified:   analysis/Fbhighres.Rmd
    Modified:   analysis/Merge-QC-UMAPS.Rmd
    Deleted:    analysis/markerGenesRNAAllFb_snn_res.0.25
    Deleted:    analysis/markerGenesRNA_snn_res.0.4

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Fbhighres.Rmd</code>) and HTML
(<code>docs/Fbhighres.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/angeldemartin/HumanHeartCarTrans2/blob/a9a9162055dbf59e8833298a961a773e821f2fff/analysis/Fbhighres.Rmd" target="_blank">a9a9162</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-07-02
</td>
<td>
july02
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/angeldemartin/HumanHeartCarTrans2/a9a9162055dbf59e8833298a961a773e821f2fff/docs/Fbhighres.html" target="_blank">a9a9162</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-07-02
</td>
<td>
july02
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/angeldemartin/HumanHeartCarTrans2/blob/f1b5059e279271f77adc69bc875de045967907a0/analysis/Fbhighres.Rmd" target="_blank">f1b5059</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-22
</td>
<td>
may22
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/angeldemartin/HumanHeartCarTrans2/f1b5059e279271f77adc69bc875de045967907a0/docs/Fbhighres.html" target="_blank">f1b5059</a>
</td>
<td>
angeldemartin
</td>
<td>
2025-05-22
</td>
<td>
may22
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>load packages</h2>
</div>
<div id="load-object" class="section level2">
<h2>load object</h2>
<pre class="r"><code>fileNam &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/Human_heart_allmerged_seurat.rds&quot;
seuratM &lt;- readRDS(fileNam)</code></pre>
</div>
<div id="set-color-vectors" class="section level2">
<h2>set color vectors</h2>
<pre class="r"><code>colclusterName &lt;- c(&quot;#67001f&quot;, &quot;#f4a582&quot;,&quot;#D53E4F&quot;, &quot;#FEE08B&quot;,&quot;#8c510a&quot;,&quot;#003c30&quot;,&quot;#01665e&quot;,&quot;#66C2A5&quot;, &quot;#3288BD&quot;,&quot;#BEAED4&quot;, &quot;#c7eae5&quot;, &quot;#B09C85&quot;, &quot;#4e5a4c&quot;,&quot;#393A3F&quot;,&quot;#355C7D&quot;,&quot;#202547&quot;,&quot;#B45B5C&quot;,&quot;#725663FF&quot;,&quot;#232429&quot;,&quot;#bac1d1&quot;)
names(colclusterName) &lt;- c(&quot;CM&quot;,&quot;Fb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;,&quot;PerivFb1&quot;,&quot;PerivFb2&quot;,&quot;VSMC&quot;,&quot;BEC1&quot;,&quot;BEC2&quot;,&quot;BEC3&quot;,&quot;LEC&quot;,&quot;NC&quot;,&quot;Baror&quot;,&quot;Adipoc&quot;,&quot;Mph1&quot;,&quot;Mph2&quot;,&quot;Tcell1&quot;,&quot;Tcell2&quot;,&quot;Mastcell&quot;,&quot;Mph3&quot;)

coldiseaseCond &lt;- c(&quot;#dfc27d&quot;,&quot;#BE3144&quot;,&quot;#202547&quot;,&quot;#355C7D&quot;,&quot;#779d8d&quot;)
names(coldiseaseCond) &lt;- c(&quot;healthy&quot;, &quot;explant&quot;, &quot;visit1&quot;, &quot;visit2&quot;, &quot;visit3&quot;)</code></pre>
<pre class="r"><code>## check umap
Idents(seuratM) &lt;- seuratM$clusterName
DimPlot(seuratM, reduction = &quot;umap&quot;, cols = colclusterName, label= TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="subset-fb-perivfb-vsmc-allfb" class="section level2">
<h2>subset Fb, PerivFb, VSMC (AllFb)</h2>
<pre class="r"><code>## subset Fb, PerivFb and VSMC
seuratAllFb &lt;- subset(seuratM, idents = c(&quot;Fb1&quot;, &quot;Fb2&quot;, &quot;PerivFb1&quot;, &quot;VSMC&quot;)) ## exclude Fb3 and PerivFb2 - int cells/doublets
levels(seuratAllFb)</code></pre>
<pre><code>[1] &quot;Fb1&quot;      &quot;Fb2&quot;      &quot;PerivFb1&quot; &quot;VSMC&quot;    </code></pre>
<pre class="r"><code>table(seuratAllFb$orig.ident)</code></pre>
<pre><code>
      
83263 </code></pre>
<pre class="r"><code>table(seuratAllFb$clusterName)</code></pre>
<pre><code>
     Fb1      Fb2 PerivFb1     VSMC 
   34945    20185    24499     3634 </code></pre>
<pre class="r"><code>DimPlot(seuratAllFb, reduction = &quot;umap&quot;, cols = colclusterName, label=TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="rerun-seurat-allfb" class="section level2">
<h2>rerun seurat AllFb</h2>
<pre class="r"><code>## rerun seurat
seuratAllFb &lt;- NormalizeData (object = seuratAllFb)
seuratAllFb &lt;- FindVariableFeatures(object = seuratAllFb)
seuratAllFb &lt;- ScaleData(object = seuratAllFb, verbose = TRUE)
seuratAllFb &lt;- RunPCA(object=seuratAllFb, npcs = 30, verbose = FALSE)
seuratAllFb &lt;- RunTSNE(object=seuratAllFb, reduction=&quot;pca&quot;, dims = 1:20)
seuratAllFb &lt;- RunUMAP(object=seuratAllFb, reduction=&quot;pca&quot;, dims = 1:20)
seuratAllFb &lt;- FindNeighbors(object = seuratAllFb, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4)
for (i in 1:length(res)) {
  seuratAllFb &lt;- FindClusters(object = seuratAllFb, resolution = res[i], random.seed = 1234)
}

table(seuratAllFb$orig.ident)
table(seuratAllFb$RNA_snn_res.0.25)</code></pre>
</div>
<div id="load-file-allfb" class="section level2">
<h2>load file AllFb</h2>
<pre class="r"><code>fileNam &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/Human_heart_AllFb.rds&quot;
seuratAllFb &lt;- readRDS(fileNam)
table(seuratAllFb$dataset)</code></pre>
<pre><code>
        331571_3-5_20231012_Hu_nucseq_USZ_HTx001         331571_4-6_20231012_Hu_nucseq_USZ_HTx002 
                                            1792                                             1211 
    334131_01-1_20231103_Hu_nucseq_USZ_HTx002_RV     334131_02-2_20231103_Hu_nucseq_USZ_HTx003_LV 
                                            2352                                             1649 
    334131_03-3_20231103_Hu_nucseq_USZ_HTx004_RV     334131_04-4_20231106_Hu_nucseq_USZ_HTx003_RV 
                                            1917                                             1086 
    334131_05-5_20231106_Hu_nucseq_USZ_HTx004_LV     334131_06-6_20231106_Hu_nucseq_USZ_HTx005_RV 
                                            2667                                             1404 
    334131_07-7_20231107_Hu_nucseq_USZ_HTx005_LV     334131_08-8_20231107_Hu_nucseq_USZ_HTx006_RV 
                                             752                                             1190 
    334131_09-9_20231108_Hu_nucseq_USZ_HTx006_LV    334131_10-10_20231108_Hu_nucseq_USZ_HTx001_RV 
                                            2226                                             1395 
    336431_07-7_20231129_Hu_nucseq_USZ_HTx008_RV     336431_08-8_20231129_Hu_nucseq_USZ_EMB001_V1 
                                            2598                                              267 
    336431_09-9_20231129_Hu_nucseq_USZ_HTx008_LV    336431_10-10_20231129_Hu_nucseq_USZ_EMB005_V1 
                                            2168                                              451 
   336431_11-11_20231129_Hu_nucseq_USZ_EMB006_V1    336431_12-12_20231129_Hu_nucseq_USZ_EMB007_V1 
                                             299                                              685 
   336431_13-13_20231129_Hu_nucseq_USZ_EMB002_V1    336431_14-14_20231129_Hu_nucseq_USZ_EMB003_V1 
                                             403                                              886 
   336431_15-15_20231129_Hu_nucseq_USZ_EMB004_V1     340821_01-7_20240118_Hu_nucseq_USZ_EMB007_V2 
                                             445                                              689 
    340821_02-8_20240118_Hu_nucseq_USZ_EMB008_V2     340821_03-9_20240123_Hu_nucseq_USZ_EMB001_V3 
                                             323                                               37 
   340821_04-10_20240123_Hu_nucseq_USZ_EMB002_V3    340821_05-11_20240123_Hu_nucseq_USZ_EMB003_V3 
                                             239                                               80 
   340821_06-12_20240123_Hu_nucseq_USZ_EMB004_V3    340821_07-13_20240123_Hu_nucseq_USZ_EMB005_V3 
                                             295                                             1178 
   340821_08-14_20240123_Hu_nucseq_USZ_EMB006_V3    340821_09-15_20240123_Hu_nucseq_USZ_EMB007_V3 
                                             135                                              202 
   340821_10-16_20240123_Hu_nucseq_USZ_EMB008_V3    340821_11-17_20240123_Hu_nucseq_USZ_EMB008_V1 
                                             109                                               25 
   340821_12-18_20240123_Hu_nucseq_USZ_HTx007_RV    340821_13-19_20240123_Hu_nucseq_USZ_HTx007_LV 
                                            1876                                             1428 
     340831_1-1_20240118_Hu_nucseq_USZ_EMB001_V2      340831_2-2_20240118_Hu_nucseq_USZ_EMB002_V2 
                                             481                                             1155 
     340831_3-3_20240118_Hu_nucseq_USZ_EMB003_V2      340831_4-4_20240118_Hu_nucseq_USZ_EMB004_V2 
                                             140                                              280 
     340831_5-5_20240118_Hu_nucseq_USZ_EMB005_V2      340831_6-6_20240118_Hu_nucseq_USZ_EMB006_V2 
                                             231                                              492 
     347741_4-4_20240326_Hu_nucseq_USZ_HTx010_RV      347741_5-5_20240326_Hu_nucseq_USZ_HTx010_LV 
                                             899                                             1882 
    353921_01-1_20240515_Hu_nucseq_USZ_HTx011_RV     353921_02-2_20240515_Hu_nucseq_USZ_HTx011_LV 
                                             859                                              578 
    353921_03-3_20240515_Hu_nucseq_USZ_EMB011_V1     353921_04-4_20240515_Hu_nucseq_USZ_EMB011_V2 
                                             355                                              104 
    353921_05-5_20240515_Hu_nucseq_USZ_HTx012_RV     353921_06-6_20240515_Hu_nucseq_USZ_HTx012_LV 
                                             494                                             1299 
    353921_07-7_20240515_Hu_nucseq_USZ_EMB012_V1     353921_08-8_20240515_Hu_nucseq_USZ_EMB010_V3 
                                             299                                              237 
    353921_09-9_20240515_Hu_nucseq_USZ_HTx013_RV    353921_10-10_20240515_Hu_nucseq_USZ_HTx013_LV 
                                             597                                              802 
 353921_11-11_20240515_Hu_nucseq_USZ_EMB010_V1_1  353921_12-12_20240515_Hu_nucseq_USZ_EMB010_V1_2 
                                             420                                              110 
   353921_13-13_20240524_Hu_nucseq_USZ_HTx015_RV    353921_14-14_20240524_Hu_nucseq_USZ_EMB011_V3 
                                            1060                                              304 
   353921_15-15_20240524_Hu_nucseq_USZ_EMB012_V2    353921_17-17_20240524_Hu_nucseq_USZ_HTx015_LV 
                                            1075                                             1049 
   353921_18-18_20240524_Hu_nucseq_USZ_EMB013_V1    353921_19-19_20240524_Hu_nucseq_USZ_EMB013_V2 
                                             144                                               65 
   353921_20-20_20240524_Hu_nucseq_USZ_EMB013_V3    353921_21-21_20240524_Hu_nucseq_USZ_HTx014_RV 
                                              94                                             1764 
   353921_22-22_20240524_Hu_nucseq_USZ_HTx014_LV    353921_23-23_20240524_Hu_nucseq_USZ_EMB014_V1 
                                            1619                                              403 
   353921_24-24_20240524_Hu_nucseq_USZ_EMB014_V2     359861_01-1_20240715_Hu_nucseq_USZ_HTx016_LV 
                                             172                                             3900 
    359861_02-2_20240715_Hu_nucseq_USZ_HTx016_RV     359861_03-3_20240715_Hu_nucseq_USZ_EMB014_V3 
                                            4775                                              199 
    359861_04-4_20240715_Hu_nucseq_USZ_EMB015_V1     359861_05-5_20240715_Hu_nucseq_USZ_EMB015_V2 
                                              42                                               41 
    359861_06-6_20240715_Hu_nucseq_USZ_EMB015_V3     359861_07-7_20240715_Hu_nucseq_USZ_EMB016_V1 
                                             255                                              275 
    359861_08-8_20240715_Hu_nucseq_USZ_EMB016_V2    359861_18-10_20240730_Hu_nucseq_USZ_EMB016_V3 
                                              36                                              136 
 373291_14-6_20241212_Hu_nucseq_USZ_HTx018_expRV  373291_15-7_20241212_Hu_nucseq_USZ_HTx018_expLV 
                                             159                                              212 
    373291_16-8_20241212_Hu_nucseq_USZ_HTx018_V1     373291_17-9_20241212_Hu_nucseq_USZ_HTx018_V2 
                                              20                                               55 
   373291_18-10_20241212_Hu_nucseq_USZ_HTx018_V3   o28576_1_08-8_20220525_Hu_nucseq_Graz_8_HH_GEM 
                                               7                                             1872 
o28576_1_10-10_20220525_Hu_nucseq_Graz_10_HH_GEM o28576_1_11-11_20220525_Hu_nucseq_Graz_11_HH_GEM 
                                            1615                                             2065 
o28576_1_12-12_20220525_Hu_nucseq_Graz_12_HH_GEM     o292731_1-1_20220818_Hu_nucseq_Graz_9_HH_GEM 
                                            2176                                             3042 
   o292731_2-2_20220818_Hu_nucseq_Graz_13_HH_GEM   o294781_01-1_20220912_Hu_nucseq_Graz_21_HH_GEM 
                                            4846                                              888 
  o294781_02-2_20220912_Hu_nucseq_Graz_22_HH_GEM   o294781_03-3_20220912_Hu_nucseq_Graz_23_HH_GEM 
                                            1046                                              403 
  o294781_04-4_20220912_Hu_nucseq_Graz_24_HH_GEM 
                                            1276 </code></pre>
<pre class="r"><code>table(seuratAllFb$RNA_snn_res.0.25)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7 
17372 16705 13608 13313 11627  6368  3164  1106 </code></pre>
<pre class="r"><code>table(seuratAllFb$orig.ident)</code></pre>
<pre><code>
      
83263 </code></pre>
<pre class="r"><code>#### cluster_name_Fb
seuratAllFb$clusterNameAllFb &lt;- &quot;clusterNameAllFb&quot;
seuratAllFb$clusterNameAllFb[which(seuratAllFb$RNA_snn_res.0.25 %in% &quot;0&quot; )] &lt;- &quot;Fb1&quot;
seuratAllFb$clusterNameAllFb[which(seuratAllFb$RNA_snn_res.0.25 %in% &quot;1&quot; )] &lt;- &quot;Fb2&quot;
seuratAllFb$clusterNameAllFb[which(seuratAllFb$RNA_snn_res.0.25 %in% &quot;2&quot; )] &lt;- &quot;Fb3&quot;
seuratAllFb$clusterNameAllFb[which(seuratAllFb$RNA_snn_res.0.25 %in% &quot;3&quot; )] &lt;- &quot;PerivFb1&quot;
seuratAllFb$clusterNameAllFb[which(seuratAllFb$RNA_snn_res.0.25 %in% &quot;4&quot; )] &lt;- &quot;PerivFb2&quot;
seuratAllFb$clusterNameAllFb[which(seuratAllFb$RNA_snn_res.0.25 %in% &quot;5&quot; )] &lt;- &quot;Fb4&quot;
seuratAllFb$clusterNameAllFb[which(seuratAllFb$RNA_snn_res.0.25 %in% &quot;6&quot; )] &lt;- &quot;VSMC&quot;
seuratAllFb$clusterNameAllFb[which(seuratAllFb$RNA_snn_res.0.25 %in% &quot;7&quot; )] &lt;- &quot;Fb5&quot;

table(seuratAllFb$clusterNameAllFb)
table(seuratAllFb$RNA_snn_res.0.25)

###order
Idents(seuratAllFb) &lt;- seuratAllFb$clusterNameAllFb
seuratAllFb$clusterNameAllFb &lt;- factor(seuratAllFb$clusterNameAllFb, levels=c(&quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;Fb4&quot;, &quot;Fb5&quot;, &quot;PerivFb1&quot;, &quot;PerivFb2&quot;, &quot;VSMC&quot;))
Idents(seuratAllFb) &lt;- seuratAllFb$clusterNameAllFb
table(seuratAllFb$clusterNameAllFb)</code></pre>
<pre class="r"><code>### save seurat object
saveRDS(seuratAllFb, file=&quot;/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/Human_heart_AllFb.rds&quot;)</code></pre>
<p>##set color vectors AllFb</p>
<pre class="r"><code>colAllFb &lt;- c(&quot;#D53E4F&quot;,&quot;#f4a582&quot;,&quot;#ff7b7b&quot;,&quot;#8e0b00&quot;,&quot;#FEE08B&quot;,&quot;#42090D&quot;,&quot;#FF7B00&quot;,&quot;#FFF4DF&quot;)
names(colAllFb) &lt;- c(&quot;Fb1&quot;,&quot;PerivFb1&quot;,&quot;Fb2&quot;,&quot;Fb3&quot;,&quot;VSMC&quot;,&quot;Fb4&quot;,&quot;PerivFb2&quot;, &quot;Fb5&quot;)</code></pre>
</div>
<div id="umaps" class="section level2">
<h2>umaps</h2>
<pre class="r"><code>Idents(seuratAllFb) &lt;- seuratAllFb$RNA_snn_res.0.25
DimPlot(seuratAllFb, reduction = &quot;umap&quot;, pt.size = 0.5, label = TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20AllFb-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratAllFb) &lt;- seuratAllFb$clusterName
DimPlot(seuratAllFb, reduction = &quot;umap&quot;, pt.size = 0.5, cols = colclusterName, label = TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20AllFb-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratAllFb) &lt;- seuratAllFb$clusterNameAllFb
DimPlot(seuratAllFb, reduction = &quot;umap&quot;, pt.size = 0.5, cols = colAllFb, label = TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20AllFb-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuratAllFb, reduction = &quot;umap&quot;, pt.size = 0.5, cols = colAllFb)  + theme(legend.position = &quot;null&quot;)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20AllFb-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratAllFb) &lt;- seuratAllFb$diseaseCond
DimPlot(seuratAllFb, reduction = &quot;umap&quot;, pt.size = 0.5, cols = coldiseaseCond, shuffle= TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20AllFb-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(seuratAllFb, reduction = &quot;umap&quot;, pt.size = 0.5, cols = coldiseaseCond, shuffle = TRUE) + theme(legend.position = &quot;null&quot;)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20AllFb-6.png" width="672" style="display: block; margin: auto;" />
## calculate cluster marker genes AllFb</p>
<pre class="r"><code>## cluster marker
Idents(seuratAllFb) &lt;- seuratAllFb$clusterNameAllFb
markerGenes &lt;- FindAllMarkers(seuratAllFb, only.pos=T) %&gt;% 
  dplyr::filter(p_val_adj &lt; 0.01)

## save table
write.table(markerGenes, 
            file= &quot;/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/markerGenesAllFb_clusterNameAllFb&quot;,
            sep=&quot;\t&quot;,
            quote=F,
            row.names=F,
            col.names=T)</code></pre>
<div id="bmp-features" class="section level3">
<h3>BMP features</h3>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratAllFb)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=c(&quot;HGF&quot;, &quot;MET&quot;, &quot;GREM1&quot;, &quot;GREM2&quot;, &quot;BMPR1A&quot;, &quot;BMPR2&quot;, &quot;BMP2K&quot;, &quot;BMP8A&quot;, &quot;BMP1&quot;, &quot;BMP6&quot;, &quot;BMP2&quot;, &quot;BMP5&quot;, &quot;BMP4&quot;)) %&gt;% 
  left_join(., genes, by=&quot;geneID&quot;)

pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- FeaturePlot(seuratAllFb, features = x, 
                   reduction = &quot;umap&quot;, pt.size = 0.1, 
                   cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), 
                   raster = FALSE) +
    theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
<p><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20AllFb-13.png" width="672" style="display: block; margin: auto;" />
### BMP featuresb ordered</p>
<pre class="r"><code>genes &lt;- data.frame(gene=rownames(seuratAllFb)) %&gt;% 
  mutate(geneID=gsub(&quot;^.*\\.&quot;, &quot;&quot;, gene))

selGenes &lt;- data.frame(geneID=c(&quot;HGF&quot;, &quot;MET&quot;, &quot;GREM1&quot;, &quot;GREM2&quot;, &quot;BMPR1A&quot;, &quot;BMPR2&quot;, &quot;BMP2K&quot;, &quot;BMP8A&quot;, &quot;BMP1&quot;, &quot;BMP6&quot;, &quot;BMP2&quot;, &quot;BMP5&quot;, &quot;BMP4&quot;)) %&gt;% 
  left_join(., genes, by=&quot;geneID&quot;)

pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- FeaturePlot(seuratAllFb, features = x, 
                   reduction = &quot;umap&quot;, pt.size = 0.1, 
                   cols = c(&quot;lightgrey&quot;, &quot;#BE3144&quot;), 
                   raster = FALSE, 
                   order=TRUE) +
    theme(legend.position=&quot;right&quot;)
  plot(p)
})</code></pre>
<p><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/features%20BMP%20ordered%20AllFb-13.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="bmp-violin-plots" class="section level3">
<h3>BMP violin plots</h3>
<pre class="r"><code>Idents(seuratAllFb) &lt;- seuratAllFb$diseaseCond
pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- VlnPlot( object = seuratAllFb, features = x, pt.size = 1, cols=coldiseaseCond)
  plot(p)
})</code></pre>
<p><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20BMP%20AllFb-13.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="violin-plots" class="section level3">
<h3>violin plots</h3>
<pre class="r"><code>Idents(seuratAllFb) &lt;- seuratAllFb$diseaseCond
pList &lt;- sapply(selGenes$gene, function(x){
  p &lt;- VlnPlot( object = seuratAllFb, features = x, pt.size = 0, cols=coldiseaseCond)
  plot(p)
})</code></pre>
<p><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-5.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-6.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-7.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-8.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-9.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-10.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-11.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-12.png" width="672" style="display: block; margin: auto;" /><img src="figure/Fbhighres.Rmd/violin%20plots%20AllFb-13.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="abundance-plots" class="section level2">
<h2>abundance plots</h2>
<div id="dataset" class="section level3">
<h3>dataset</h3>
<pre class="r"><code>order_keywords &lt;- c(&quot;HH&quot;, &quot;RV|LV|expLV|expRV|331571_3-5_20231012_Hu_nucseq_USZ_HTx001|331571_4-6_20231012_Hu_nucseq_USZ_HTx002&quot;, &quot;V1&quot;, &quot;V2&quot;,&quot;V3&quot;)
files &lt;- unique(seuratAllFb$dataset)
ordered_files &lt;- c()
for (key in order_keywords) {
  ordered_files &lt;- c(ordered_files, files[grepl(key, files)])
}

###dataset
datList &lt;- NULL
for(con in unique(seuratAllFb$dataset)){
  seuratSub &lt;- subset(seuratAllFb, dataset==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterNameAllFb)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(dataset=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 45337  1792
[1] 45337  1211
[1] 45337  2352
[1] 45337  1649
[1] 45337  1917
[1] 45337  1086
[1] 45337  2667
[1] 45337  1404
[1] 45337   752
[1] 45337  1190
[1] 45337  2226
[1] 45337  1395
[1] 45337  2598
[1] 45337   267
[1] 45337  2168
[1] 45337   451
[1] 45337   299
[1] 45337   685
[1] 45337   403
[1] 45337   886
[1] 45337   445
[1] 45337   689
[1] 45337   323
[1] 45337    37
[1] 45337   239
[1] 45337    80
[1] 45337   295
[1] 45337  1178
[1] 45337   135
[1] 45337   202
[1] 45337   109
[1] 45337    25
[1] 45337  1876
[1] 45337  1428
[1] 45337   481
[1] 45337  1155
[1] 45337   140
[1] 45337   280
[1] 45337   231
[1] 45337   492
[1] 45337   899
[1] 45337  1882
[1] 45337   859
[1] 45337   578
[1] 45337   355
[1] 45337   104
[1] 45337   494
[1] 45337  1299
[1] 45337   299
[1] 45337   237
[1] 45337   597
[1] 45337   802
[1] 45337   420
[1] 45337   110
[1] 45337  1060
[1] 45337   304
[1] 45337  1075
[1] 45337  1049
[1] 45337   144
[1] 45337    65
[1] 45337    94
[1] 45337  1764
[1] 45337  1619
[1] 45337   403
[1] 45337   172
[1] 45337  3900
[1] 45337  4775
[1] 45337   199
[1] 45337    42
[1] 45337    41
[1] 45337   255
[1] 45337   275
[1] 45337    36
[1] 45337   136
[1] 45337   159
[1] 45337   212
[1] 45337    20
[1] 45337    55
[1] 45337     7
[1] 45337  1872
[1] 45337  1615
[1] 45337  2065
[1] 45337  2176
[1] 45337  3042
[1] 45337  4846
[1] 45337   888
[1] 45337  1046
[1] 45337   403
[1] 45337  1276</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

## plot abundance
ggbarplot(dat_all, x= &quot;dataset&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colAllFb, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_x_discrete(limits=ordered_files)</code></pre>
<p><img src="figure/Fbhighres.Rmd/abundance%20dataset%20AllFb-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="patient_diseasecond" class="section level3">
<h3>patient_diseaseCond</h3>
<pre class="r"><code>order_keywords &lt;- c(&quot;healthy&quot;, &quot;visit1&quot;, &quot;visit2&quot;, &quot;visit3&quot;, &quot;explant&quot;)
files &lt;- unique(seuratAllFb$patient_diseaseCond)
ordered_files &lt;- c()
for (key in order_keywords) {
  ordered_files &lt;- c(ordered_files, files[grepl(key, files)])
}

###patient_diseaseCond
datList &lt;- NULL
for(con in unique(seuratAllFb$patient_diseaseCond)){
  seuratSub &lt;- subset(seuratAllFb, patient_diseaseCond==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterNameAllFb)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(patient_diseaseCond=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 45337  3187
[1] 45337  3563
[1] 45337  2735
[1] 45337  4584
[1] 45337  2156
[1] 45337  3416
[1] 45337  4766
[1] 45337   267
[1] 45337   451
[1] 45337   299
[1] 45337   685
[1] 45337   403
[1] 45337   886
[1] 45337   445
[1] 45337   689
[1] 45337   323
[1] 45337    37
[1] 45337   239
[1] 45337    80
[1] 45337   295
[1] 45337  1178
[1] 45337   135
[1] 45337   202
[1] 45337   109
[1] 45337    25
[1] 45337  3304
[1] 45337   481
[1] 45337  1155
[1] 45337   140
[1] 45337   280
[1] 45337   231
[1] 45337   492
[1] 45337  2781
[1] 45337  1437
[1] 45337   355
[1] 45337   104
[1] 45337  1793
[1] 45337   299
[1] 45337   237
[1] 45337  1399
[1] 45337   420
[1] 45337   110
[1] 45337  2109
[1] 45337   304
[1] 45337  1075
[1] 45337   144
[1] 45337    65
[1] 45337    94
[1] 45337  3383
[1] 45337   403
[1] 45337   172
[1] 45337  8675
[1] 45337   199
[1] 45337    42
[1] 45337    41
[1] 45337   255
[1] 45337   275
[1] 45337    36
[1] 45337   136
[1] 45337   371
[1] 45337    20
[1] 45337    55
[1] 45337     7
[1] 45337  1872
[1] 45337  1615
[1] 45337  2065
[1] 45337  2176
[1] 45337  3042
[1] 45337  4846
[1] 45337   888
[1] 45337  1046
[1] 45337   403
[1] 45337  1276</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

## plot abundance
ggbarplot(dat_all, x= &quot;patient_diseaseCond&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colAllFb, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_x_discrete(limits=ordered_files)</code></pre>
<p><img src="figure/Fbhighres.Rmd/abundance%20patient_diseaseCond%20AllFb-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="diseasecond" class="section level3">
<h3>diseaseCond</h3>
<pre class="r"><code>orddiseaseCond &lt;- c(&quot;healthy&quot;,&quot;visit1&quot;, &quot;visit2&quot; ,&quot;visit3&quot;, &quot;explant&quot;)

###diseaseCond
datList &lt;- NULL
for(con in unique(seuratAllFb$diseaseCond)){
  seuratSub &lt;- subset(seuratAllFb, diseaseCond==con)
  print(dim(seuratSub))
  dat_con &lt;- as.data.frame(table(seuratSub$clusterNameAllFb)) %&gt;%
  mutate(percent=Freq/ncol(seuratSub)) %&gt;% mutate(diseaseCond=con)
  datList[[con]] &lt;- dat_con
}</code></pre>
<pre><code>[1] 45337 49659
[1] 45337  5419
[1] 45337  5449
[1] 45337  3507
[1] 45337 19229</code></pre>
<pre class="r"><code>dat_all &lt;- do.call(&quot;rbind&quot;, datList)

## plot abundance
ggbarplot(dat_all, x= &quot;diseaseCond&quot;, y= &quot;percent&quot;, fill = &quot;Var1&quot;, palette = colAllFb, legend = &quot;right&quot;, legend.titel = &quot;cluster&quot;, ylab = &quot;frequency&quot;)  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + scale_x_discrete(limits=orddiseaseCond)</code></pre>
<p><img src="figure/Fbhighres.Rmd/abundance%20diseaseCoond%20AllFb-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="subset-intfb" class="section level2">
<h2>subset IntFb</h2>
<pre class="r"><code>## subset IntFb
Idents(seuratAllFb) &lt;- seuratAllFb$clusterNameAllFb
seuratIntFb &lt;- subset(seuratAllFb, idents = c(&quot;Fb1&quot;, &quot;Fb2&quot;, &quot;Fb3&quot;, &quot;Fb4&quot;, &quot;Fb5&quot;))
table(seuratIntFb$clusterNameAllFb)</code></pre>
<pre><code>
  Fb1   Fb2   Fb3   Fb4   Fb5 
17372 16705 13608  6368  1106 </code></pre>
<pre class="r"><code>DimPlot(seuratIntFb, reduction = &quot;umap&quot;, cols = colAllFb, label=TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="rerun-seurat-intfb" class="section level2">
<h2>rerun seurat IntFb</h2>
<pre class="r"><code>## rerun seurat
seuratIntFb &lt;- NormalizeData (object = seuratIntFb)
seuratIntFb &lt;- FindVariableFeatures(object = seuratIntFb)
seuratIntFb &lt;- ScaleData(object = seuratIntFb, verbose = TRUE)
seuratIntFb &lt;- RunPCA(object=seuratIntFb, npcs = 30, verbose = FALSE)
seuratIntFb &lt;- RunTSNE(object=seuratIntFb, reduction=&quot;pca&quot;, dims = 1:20)
seuratIntFb &lt;- RunUMAP(object=seuratIntFb, reduction=&quot;pca&quot;, dims = 1:20)
seuratIntFb &lt;- FindNeighbors(object = seuratIntFb, reduction = &quot;pca&quot;, dims= 1:20)

res &lt;- c(0.25, 0.6, 0.8, 0.4, 0.1)
for (i in 1:length(res)) {
  seuratIntFb &lt;- FindClusters(object = seuratIntFb, resolution = res[i], random.seed = 1234)
}

table(seuratIntFb$orig.ident)
table(seuratIntFb$RNA_snn_res.0.25)</code></pre>
<pre class="r"><code>## clustername IntFb
seuratIntFb$clusterNameIntFb &lt;- &quot;clusterNameIntFb&quot;
seuratIntFb$clusterNameIntFb[which(seuratIntFb$RNA_snn_res.0.1 %in% &quot;0&quot; )] &lt;- &quot;Fb1/2&quot;
seuratIntFb$clusterNameIntFb[which(seuratIntFb$RNA_snn_res.0.1 %in% &quot;1&quot; )] &lt;- &quot;Fb3&quot;
seuratIntFb$clusterNameIntFb[which(seuratIntFb$RNA_snn_res.0.1 %in% &quot;2&quot; )] &lt;- &quot;Fb4&quot;
seuratIntFb$clusterNameIntFb[which(seuratIntFb$RNA_snn_res.0.1 %in% &quot;3&quot; )] &lt;- &quot;Fb5&quot;

table(seuratIntFb$clusterNameIntFb)
table(seuratIntFb$RNA_snn_res.0.1)

## order
Idents(seuratIntFb) &lt;- seuratIntFb$clusterNameIntFb
seuratIntFb$clusterNameIntFb &lt;- factor(seuratIntFb$clusterNameIntFb, levels=c(&quot;Fb1/2&quot;, &quot;Fb3&quot;, &quot;Fb4&quot;, &quot;Fb5&quot;))
Idents(seuratIntFb) &lt;- seuratIntFb$clusterNameIntFb
table(seuratIntFb$clusterNameIntFb)</code></pre>
<pre class="r"><code>### save seurat object
saveRDS(seuratIntFb, file=&quot;/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/Human_heart_IntFb.rds&quot;)</code></pre>
</div>
<div id="load-file-intfb" class="section level2">
<h2>load file IntFb</h2>
<pre class="r"><code>fileNam &lt;- &quot;/Users/immbio/Desktop/Project/Angelina/HumanHeartCarTrans2/data/Human_heart_IntFb.rds&quot;
seuratIntFb &lt;- readRDS(fileNam)
table(seuratIntFb$clusterNameIntFb, seuratIntFb$diseaseCond)</code></pre>
<pre><code>       
        healthy visit1 visit2 visit3 explant
  Fb1/2     643    405    548    274   32433
  Fb3       749   3744   3546   2398    2332
  Fb4      6935     16     54     27     117
  Fb5       924      7      2      5       0</code></pre>
<pre class="r"><code>colIntFb &lt;- c(&quot;#D53E4F&quot;,&quot;#8e0b00&quot;,&quot;#42090D&quot;,&quot;#FFF4DF&quot;)
names(colIntFb) &lt;- c(&quot;Fb1/2&quot;,&quot;Fb3&quot;,&quot;Fb4&quot;, &quot;Fb5&quot;)</code></pre>
</div>
<div id="umaps-1" class="section level2">
<h2>umaps</h2>
<pre class="r"><code>Idents(seuratIntFb) &lt;- seuratIntFb$RNA_snn_res.0.1
DimPlot(seuratIntFb, reduction = &quot;umap&quot;, pt.size = 0.5, label = TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20IntFb-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratIntFb) &lt;- seuratIntFb$clusterNameAllFb
DimPlot(seuratIntFb, reduction = &quot;umap&quot;, pt.size = 0.5, cols = colAllFb, label = TRUE, shuffle = TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20IntFb-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratIntFb) &lt;- seuratIntFb$clusterNameIntFb
DimPlot(seuratIntFb, reduction = &quot;umap&quot;, pt.size = 0.5, cols = colIntFb, label = TRUE, shuffle = TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20IntFb-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(seuratIntFb) &lt;- seuratIntFb$diseaseCond
DimPlot(seuratIntFb, reduction = &quot;umap&quot;, pt.size = 0.5, cols = coldiseaseCond, shuffle = TRUE)</code></pre>
<p><img src="figure/Fbhighres.Rmd/umaps%20IntFb-4.png" width="672" style="display: block; margin: auto;" /></p>
<div id="diffusion-map" class="section level3">
<h3>diffusion map</h3>
<pre class="r"><code>pca_embeddings &lt;- Embeddings(seuratIntFb, reduction = &quot;pca&quot;)
dm &lt;- DiffusionMap(pca_embeddings)

df &lt;- data.frame(DC1 = eigenvectors(dm)[,1], DC2 = eigenvectors(dm)[,2])

## color according to clusterAllFb
df$cluster &lt;- seuratIntFb$clusterNameAllFb

ggplot(df, aes(x = DC1, y = DC2, color = cluster)) +
  geom_point(size = 0.5) +
  scale_color_manual(values = colAllFb)</code></pre>
<p><img src="figure/Fbhighres.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## color according to diseaseCond
df$cluster &lt;- seuratIntFb$diseaseCond

ggplot(df, aes(x = DC1, y = DC2, color = cluster)) +
  geom_point(size = 0.5) +
  scale_color_manual(values = coldiseaseCond)</code></pre>
<p><img src="figure/Fbhighres.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="session-info" class="section level2">
<h2>session info</h2>
<pre class="r"><code>date()</code></pre>
<pre><code>[1] &quot;Thu Jul 24 16:06:40 2025&quot;</code></pre>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
 [4] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
 [7] enrichplot_1.24.4           msigdbr_24.1.0              org.Hs.eg.db_3.19.1        
[10] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[13] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[16] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[19] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[22] forcats_1.0.0               stringr_1.5.1               purrr_1.0.4                
[25] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[28] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[31] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[34] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[37] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[40] pheatmap_1.0.13             ggpubr_0.6.0                ggplot2_3.5.2              
[43] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[46] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.3                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] knn.covertree_1.0         ggtree_3.12.0             data.table_1.17.4        
 [79] R.oo_1.27.1               RSpectra_0.16-2           irlba_2.3.5.1            
 [82] ggrastr_1.0.2             gridGraphics_0.5-1        fastDummies_1.7.5        
 [85] lazyeval_0.2.2            yaml_2.3.10               survival_3.8-3           
 [88] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [91] RColorBrewer_1.1-3        progressr_0.15.1          tweenr_2.0.3             
 [94] later_1.4.2               ggridges_0.5.6            codetools_0.2-20         
 [97] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
[100] Rtsne_0.17                shape_1.4.6.1             limma_3.60.6             
[103] pkgconfig_2.0.3           TMB_1.9.17                spatstat.univar_3.1-3    
[106] mathjaxr_1.8-0            EnvStats_3.1.0            aplot_0.2.5              
[109] scatterplot3d_0.3-44      ape_5.8-1                 spatstat.sparse_3.1-0    
[112] xtable_1.8-4              car_3.1-3                 plyr_1.8.9               
[115] httr_1.4.7                rbibutils_2.3             tools_4.4.0              
[118] globals_0.18.0            beeswarm_0.4.0            broom_1.0.8              
[121] nlme_3.1-168              lambda.r_1.2.4            assertthat_0.2.1         
[124] lme4_1.1-37               digest_0.6.37             numDeriv_2016.8-1.1      
[127] Matrix_1.7-3              farver_2.1.2              tzdb_0.5.0               
[130] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.2.0        
[133] glue_1.8.0                cachem_1.1.0              polyclip_1.10-7          
[136] generics_0.1.4            Biostrings_2.72.1         mvtnorm_1.3-3            
[139] parallelly_1.45.0         mnormt_2.1.1              statmod_1.5.0            
[142] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[145] minqa_1.2.8               pbapply_1.7-2             httr2_1.1.2              
[148] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[151] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[154] shiny_1.10.0              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[157] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[160] scales_1.4.0              R.methodsS3_1.8.2         future_1.58.0            
[163] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-6      
[166] rstudioapi_0.17.1         cluster_2.1.8.1           whisker_0.4.1            
[169] mutoss_0.1-13             spatstat.utils_3.1-4      hms_1.1.3                
[172] fitdistrplus_1.2-2        cowplot_1.1.3             colorspace_2.1-1         
[175] rlang_1.1.6               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[178] xts_0.14.1                dotCall64_1.2             shinydashboard_0.7.3     
[181] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-3               
[184] xfun_0.52                 e1071_1.7-16              TH.data_1.1-3            
[187] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[190] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[193] Rdpack_2.6.4              promises_1.3.3            scatterpie_0.2.4         
[196] RSQLite_2.4.0             qvalue_2.36.0             sandwich_3.1-1           
[199] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[202] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[205] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[208] Rcpp_1.0.14               edgeR_4.2.2               workflowr_1.7.1          
[211] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-65              
[214] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[217] spatstat.random_3.4-1     R6_2.6.1                  fastmap_1.2.0            
[220] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[223] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[226] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[229] nnet_7.3-20               gtable_0.3.6              KernSmooth_2.23-26       
[232] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[235] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.4-3   
[238] lifecycle_1.0.4           blme_1.0-6                nloptr_2.2.1             
[241] sass_0.4.10               vctrs_0.6.5               robustbase_0.99-4-1      
[244] spatstat.geom_3.4-1       sn_2.1.1                  ggfun_0.1.8              
[247] future.apply_1.11.3       bslib_0.9.0               pillar_1.10.2            
[250] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.12          
[253] jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: x86_64-apple-darwin20
Running under: macOS Ventura 13.7.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] NCmisc_1.2.0                VennDiagram_1.7.3           futile.logger_1.4.3        
 [4] ggupset_0.4.1               gridExtra_2.3               DOSE_3.30.5                
 [7] enrichplot_1.24.4           msigdbr_24.1.0              org.Hs.eg.db_3.19.1        
[10] AnnotationDbi_1.66.0        clusterProfiler_4.12.6      multtest_2.60.0            
[13] metap_1.12                  scater_1.32.1               scuttle_1.14.0             
[16] destiny_3.18.0              circlize_0.4.16             muscat_1.18.0              
[19] viridis_0.6.5               viridisLite_0.4.2           lubridate_1.9.4            
[22] forcats_1.0.0               stringr_1.5.1               purrr_1.0.4                
[25] readr_2.1.5                 tidyr_1.3.1                 tibble_3.2.1               
[28] tidyverse_2.0.0             dplyr_1.1.4                 SingleCellExperiment_1.26.0
[31] SummarizedExperiment_1.34.0 Biobase_2.64.0              GenomicRanges_1.56.2       
[34] GenomeInfoDb_1.40.1         IRanges_2.38.1              S4Vectors_0.42.1           
[37] BiocGenerics_0.50.0         MatrixGenerics_1.16.0       matrixStats_1.5.0          
[40] pheatmap_1.0.13             ggpubr_0.6.0                ggplot2_3.5.2              
[43] Seurat_5.3.0                SeuratObject_5.1.0          sp_2.2-0                   
[46] runSeurat3_0.1.0            ExploreSCdataSeurat3_0.1.0 

loaded via a namespace (and not attached):
  [1] igraph_2.1.4              ica_1.0-3                 plotly_4.10.4            
  [4] Formula_1.2-5             zlibbioc_1.50.0           tidyselect_1.2.1         
  [7] bit_4.6.0                 doParallel_1.0.17         clue_0.3-66              
 [10] lattice_0.22-7            rjson_0.2.23              blob_1.2.4               
 [13] S4Arrays_1.4.1            pbkrtest_0.5.4            parallel_4.4.0           
 [16] png_0.1-8                 plotrix_3.8-4             cli_3.6.5                
 [19] ggplotify_0.1.2           goftest_1.2-3             VIM_6.2.2                
 [22] variancePartition_1.34.0  BiocNeighbors_1.22.0      shadowtext_0.1.4         
 [25] uwot_0.2.3                curl_6.2.3                tidytree_0.4.6           
 [28] mime_0.13                 evaluate_1.0.3            ComplexHeatmap_2.20.0    
 [31] stringi_1.8.7             backports_1.5.0           lmerTest_3.1-3           
 [34] qqconf_1.3.2              httpuv_1.6.16             magrittr_2.0.3           
 [37] rappdirs_0.3.3            splines_4.4.0             ggraph_2.2.1             
 [40] sctransform_0.4.2         ggbeeswarm_0.7.2          DBI_1.2.3                
 [43] jquerylib_0.1.4           smoother_1.3              withr_3.0.2              
 [46] git2r_0.36.2              corpcor_1.6.10            reformulas_0.4.1         
 [49] class_7.3-23              rprojroot_2.0.4           lmtest_0.9-40            
 [52] tidygraph_1.3.1           formatR_1.14              colourpicker_1.3.0       
 [55] htmlwidgets_1.6.4         fs_1.6.6                  ggrepel_0.9.6            
 [58] labeling_0.4.3            fANCOVA_0.6-1             SparseArray_1.4.8        
 [61] DESeq2_1.44.0             ranger_0.17.0             DEoptimR_1.1-3-1         
 [64] reticulate_1.42.0         hexbin_1.28.5             zoo_1.8-14               
 [67] XVector_0.44.0            knitr_1.50                ggplot.multistats_1.0.1  
 [70] UCSC.utils_1.0.0          RhpcBLASctl_0.23-42       timechange_0.3.0         
 [73] foreach_1.5.2             patchwork_1.3.0           caTools_1.18.3           
 [76] knn.covertree_1.0         ggtree_3.12.0             data.table_1.17.4        
 [79] R.oo_1.27.1               RSpectra_0.16-2           irlba_2.3.5.1            
 [82] ggrastr_1.0.2             gridGraphics_0.5-1        fastDummies_1.7.5        
 [85] lazyeval_0.2.2            yaml_2.3.10               survival_3.8-3           
 [88] scattermore_1.2           crayon_1.5.3              RcppAnnoy_0.0.22         
 [91] RColorBrewer_1.1-3        progressr_0.15.1          tweenr_2.0.3             
 [94] later_1.4.2               ggridges_0.5.6            codetools_0.2-20         
 [97] GlobalOptions_0.1.2       aod_1.3.3                 KEGGREST_1.44.1          
[100] Rtsne_0.17                shape_1.4.6.1             limma_3.60.6             
[103] pkgconfig_2.0.3           TMB_1.9.17                spatstat.univar_3.1-3    
[106] mathjaxr_1.8-0            EnvStats_3.1.0            aplot_0.2.5              
[109] scatterplot3d_0.3-44      ape_5.8-1                 spatstat.sparse_3.1-0    
[112] xtable_1.8-4              car_3.1-3                 plyr_1.8.9               
[115] httr_1.4.7                rbibutils_2.3             tools_4.4.0              
[118] globals_0.18.0            beeswarm_0.4.0            broom_1.0.8              
[121] nlme_3.1-168              lambda.r_1.2.4            assertthat_0.2.1         
[124] lme4_1.1-37               digest_0.6.37             numDeriv_2016.8-1.1      
[127] Matrix_1.7-3              farver_2.1.2              tzdb_0.5.0               
[130] remaCor_0.0.18            reshape2_1.4.4            yulab.utils_0.2.0        
[133] glue_1.8.0                cachem_1.1.0              polyclip_1.10-7          
[136] generics_0.1.4            Biostrings_2.72.1         mvtnorm_1.3-3            
[139] parallelly_1.45.0         mnormt_2.1.1              statmod_1.5.0            
[142] RcppHNSW_0.6.0            ScaledMatrix_1.12.0       carData_3.0-5            
[145] minqa_1.2.8               pbapply_1.7-2             httr2_1.1.2              
[148] spam_2.11-1               gson_0.1.0                graphlayouts_1.2.2       
[151] gtools_3.9.5              ggsignif_0.6.4            RcppEigen_0.3.4.0.2      
[154] shiny_1.10.0              GenomeInfoDbData_1.2.12   glmmTMB_1.1.11           
[157] R.utils_2.13.0            memoise_2.0.1             rmarkdown_2.29           
[160] scales_1.4.0              R.methodsS3_1.8.2         future_1.58.0            
[163] RANN_2.6.2                Cairo_1.6-2               spatstat.data_3.1-6      
[166] rstudioapi_0.17.1         cluster_2.1.8.1           whisker_0.4.1            
[169] mutoss_0.1-13             spatstat.utils_3.1-4      hms_1.1.3                
[172] fitdistrplus_1.2-2        cowplot_1.1.3             colorspace_2.1-1         
[175] rlang_1.1.6               DelayedMatrixStats_1.26.0 sparseMatrixStats_1.16.0 
[178] xts_0.14.1                dotCall64_1.2             shinydashboard_0.7.3     
[181] ggforce_0.4.2             laeken_0.5.3              mgcv_1.9-3               
[184] xfun_0.52                 e1071_1.7-16              TH.data_1.1-3            
[187] iterators_1.0.14          abind_1.4-8               GOSemSim_2.30.2          
[190] treeio_1.28.0             futile.options_1.0.1      bitops_1.0-9             
[193] Rdpack_2.6.4              promises_1.3.3            scatterpie_0.2.4         
[196] RSQLite_2.4.0             qvalue_2.36.0             sandwich_3.1-1           
[199] fgsea_1.30.0              DelayedArray_0.30.1       proxy_0.4-27             
[202] GO.db_3.19.1              compiler_4.4.0            prettyunits_1.2.0        
[205] boot_1.3-31               beachmat_2.20.0           listenv_0.9.1            
[208] Rcpp_1.0.14               edgeR_4.2.2               workflowr_1.7.1          
[211] BiocSingular_1.20.0       tensor_1.5                MASS_7.3-65              
[214] progress_1.2.3            BiocParallel_1.38.0       babelgene_22.9           
[217] spatstat.random_3.4-1     R6_2.6.1                  fastmap_1.2.0            
[220] multcomp_1.4-28           fastmatch_1.1-6           rstatix_0.7.2            
[223] vipor_0.4.7               TTR_0.24.4                ROCR_1.0-11              
[226] TFisher_0.2.0             rsvd_1.0.5                vcd_1.4-13               
[229] nnet_7.3-20               gtable_0.3.6              KernSmooth_2.23-26       
[232] miniUI_0.1.2              deldir_2.0-4              htmltools_0.5.8.1        
[235] ggthemes_5.1.0            bit64_4.6.0-1             spatstat.explore_3.4-3   
[238] lifecycle_1.0.4           blme_1.0-6                nloptr_2.2.1             
[241] sass_0.4.10               vctrs_0.6.5               robustbase_0.99-4-1      
[244] spatstat.geom_3.4-1       sn_2.1.1                  ggfun_0.1.8              
[247] future.apply_1.11.3       bslib_0.9.0               pillar_1.10.2            
[250] gplots_3.2.0              pcaMethods_1.96.0         locfit_1.5-9.12          
[253] jsonlite_2.0.0            GetoptLong_1.0.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
